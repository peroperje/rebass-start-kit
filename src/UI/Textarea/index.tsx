import * as form from "@rebass/forms";

export type TextAreaProps = form.TextareaProps;
const TextArea = form.Textarea;
export default TextArea;
