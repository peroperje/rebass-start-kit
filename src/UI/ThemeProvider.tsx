import React from "react";
import { ThemeProvider as EmotionThemeProvider } from "emotion-theming";

import theme from "./theme";

export interface Props {
  children: React.ReactNode;
}

const ThemeProvider: React.FC<Props> = ({ children }: Props): JSX.Element => (
  <EmotionThemeProvider theme={theme}>{children}</EmotionThemeProvider>
);

export default ThemeProvider;
