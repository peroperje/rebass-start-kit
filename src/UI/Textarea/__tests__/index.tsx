import React from 'react';
import render from 'react-test-renderer';

import Textarea from '../index';
import ThemeProvider from '../../ThemeProvider';

describe('Textarea', () => {
    it('Default snapshot', () => {
        const tree = render
            .create(
                <ThemeProvider>
                    <Textarea>Test</Textarea>
                </ThemeProvider>,
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
