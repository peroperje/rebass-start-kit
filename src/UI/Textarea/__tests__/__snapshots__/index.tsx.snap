// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea Default snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  display: block;
  width: 100%;
  padding: 8px;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: inherit;
  line-height: inherit;
  border: 1px solid;
  border-radius: 4px;
  color: inherit;
  background-color: transparent;
}

<textarea
  className="emotion-0"
>
  Test
</textarea>
`;
